java-logging-spring/src/test/java/uk/gov/hmcts/reform/logging/spring/RequestLoggingComponentTest.java

```diff
@@ -50,24 +52,41 @@ public void listenForEvents() throws JoranException, IOException {
     @Test
     public void requestProcessedMessageShouldBeLoggedForPublicResource() throws Exception {
         ResponseEntity<String> response = restTemplate.getForEntity("/public", String.class);
+
         assertThat(response.getBody()).isEqualTo("OK");
-        assertThat(loggedEvents()).extracting("message").containsOnlyOnce("Request processed");
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+
+        Thread.sleep(10);// how to fix this? test fails to see the log message otherwise
```

Comment:
Maybe waiting for 5 seconds until message gets logged, or fail? Have a look at http://tempusfugitlibrary.org/documentation/junit/waiting/